src/main/java/jenkins/plugins/logstash/configuration/Logzio.java

```
@@ -0,0 +1,128 @@
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+
+public class Logzio extends LogstashIndexer<LogzioDao>
+{
+    private Secret key;
+    private String host;
+
+    @DataBoundConstructor
+    public Logzio(){}
+
+    /*
+     * We use URL for the setter as stapler can autoconvert a string to a URL but not to a URI
+     */
+    public String getHost(){ return this.host; }
+
+    @DataBoundSetter
+    public void setHost(String host){ this.host = host; }
+
+    public String getKey()
+    {
+        return Secret.toString(key);
+    }
+
+    @DataBoundSetter
+    public void setKey(String key)
+    {
+        this.key = Secret.fromString(key);
+    }
+
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (obj == null)
+            return false;
+        if (this == obj)
+            return true;
+        if (getClass() != obj.getClass())
+            return false;
+        Logzio other = (Logzio) obj;
+        if (!Secret.toString(key).equals(other.getKey()))
+        {
+            return false;
+        }
+        if (host == null)
+        {
+            if (other.host != null)
+                return false;
+        }
+        else if (!host.equals(other.host))
+        {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        final int prime = 31;
+        int result = super.hashCode();
+        result = prime * result + ((host == null) ? 0 : host.hashCode());
+        result = prime * result + Secret.toString(key).hashCode();
+        return result;
+    }
+
+    @Override
+    public LogzioDao createIndexerInstance() {
+        try{
+            LogzioDao logzioDao = new LogzioDao(host, Secret.toString(key));
+            return logzioDao;
+        }catch (io.logz.sender.exceptions.LogzioParameterErrorException e){
+            return null;
```

I think we better let the `IllegalArgumentException` to be thrown instead of catching it and `return null`